Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -5368,7 +5361,9 @@ void WebGLRenderingContextBase::synthesizeLostContextGLError(GCGLenum error, ASC
 
 IntSize WebGLRenderingContextBase::clampedCanvasSize()
 {
-    IntSize canvasSize { static_cast<int>(canvasBase().width()), static_cast<int>(canvasBase().height()) };
+    auto canvasSize = canvasBase().size();
+    int maxDim = std::min(m_maxTextureSize, m_maxRenderbufferSize);
+    canvasSize.clampToMaximumSize({ maxDim, maxDim });
     return canvasSize.constrainedBetween({ 1, 1 }, { m_maxViewportDims[0], m_maxViewportDims[1] });
```

Comment:
Maybe clamping in one step might look better:
```
auto maxSize = IntSize { std::min(maxDim, m_maxViewportDims[0]), std::min(maxDim, m_maxViewportDims[1]));
return canvasSize.constrainedBetween({ 1, 1 }, maxSize);
```

Reply:
I think it's easier to understand when each constraint is in its own statement, so that there's no multiple min/max in same line.
it's like:
* clamped canvas size is
  * canvas size
  * constrained by maximum texture/renderbuffer size
  * constrained by 1,1, and maximum viewport size